pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2163,3 +2163,23 @@ def test_merge_datetime_upcast_dtype():
         }
     )
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("n_categories", [5, 128])
```

Comment:
can you add a comment here that 5 and 128 are chosen to ensure this hits specific dtypes in libjoin

Reply:
@jbrockmendel Sure, added